### PR TITLE
add docker configuratin'

### PR DESCRIPTION
**Pull request by ctb**

### CODE REVIEW

**Comment by johnsolk** [comment]
Looks like checks failed?

```
The command "literate-resting/scan.py [1-3]*.rst" exited with 0.
The command "docker run --name longy -v $(pwd):/mnt -dit diblab/eel-pond:0.1 /bin/bash" exited with 0.
The command "docker exec longy /bin/sh -c "ls -R /mnt"" exited with 0.
The command "docker exec longy /bin/sh -c "PROJECT=/mnt /mnt/1-quality.rst.sh"" exited with 126.
The command "docker exec longy /bin/sh -c "PROJECT=/mnt /mnt/2-diginorm.rst.sh"" exited with 126.
The command "docker exec longy /bin/sh -c "PROJECT=/mnt /mnt/3-big-assembly.rst.sh"" exited with 126.
```

**Comment by ctb** [comment]
yes, working on it ;)

**Comment by ctb** [comment]
Hey, look, it all works now!

**Comment by johnsolk** [comment]
Looks good to me. CI tests pass.

**Review by johnsolk — APPROVED** [review]

File: .travis.yml
Comment (standage) [inline .travis.yml:11]: I'm still new to the ways of the force, but what does this do? It looks like https://hub.docker.com/r/diblab/eel-pond/ is currently empty.

**Comment by ctb** [comment]
https://hub.docker.com/r/diblab/eel-pond/tags/

**Comment by standage** [comment]
> https://hub.docker.com/r/diblab/eel-pond/tags/

No way to inspect contents?

**Comment by ctb** [comment]
On Wed, Dec 21, 2016 at 10:54:49AM -0800, Daniel Standage wrote:
> > https://hub.docker.com/r/diblab/eel-pond/tags/
> 
> No way to inspect contents?

I... how would you do that? You mean by seeing the Dockerfile?

**Comment by standage** [comment]
> I... how would you do that? You mean by seeing the Dockerfile?

I obviously don't know enough yet to know what I don't know.

**Comment by standage** [comment]
> I obviously don't know enough yet to know what I don't know.

Maybe we could get Rumsfeld up in here.

**Comment by ctb** [comment]
On Wed, Dec 21, 2016 at 11:00:55AM -0800, Daniel Standage wrote:
> > I obviously don't know enough yet to know what I don't know.
> 
> Maybe we could get Rumsfeld up in here.

:) please no

Docker images are basically binary blobs that contain All The Things.

Dockerfiles are the recipes used to create those binary blobs.

So the former (which is what's posted to docker hub) are less interesting
than the latter.

See https://github.com/dib-lab/khmer/issues/1476#issuecomment-268409358 and
especially 
https://github.com/dib-lab/eel-pond/pull/30/files#diff-cfd2b863eadb60a3df51a7240158dba1

**Comment by standage** [comment]
Oh ok, got it. The Dockerfile was so minimal, I was wondering where all the config stuff was. But of course that's scraped out of the install docs...

...which...is...kinda the *point* of what we're doing here. :)

**Comment by ctb** [comment]
By jove, he's got it! :) :)

Yeah, I'm feeling a little wiggly about how much indirection and abstraction there is in all of this. Each individual step is simple but...

**Comment by ctb** [comment]
<ahem> please wait until there's a request for review & merge :)

**Comment by standage** [comment]
Yeah, sorry, asked on Slack but got trigger happy.

File: .travis.yml
Comment (betatim) [reply .travis.yml:11]: You can deposit the binary image produced by `docker build <blah>` on docker hub and anyone can then fetch it with `docker pull ...`. The dockerhub entry doesn't have to be linked to any github repo or similar so that is why it looks so empty over there. (You could link the dockerhub entry to a repo and then [dockerhub builds a new image for every new commit](https://docs.docker.com/docker-hub/builds/) to `master` in that repo). In that case you end up seeing the `Dockerfile` and stuff also on the dockerhub. Does that make sense?

**Pull request merged**
